src/client/Microsoft.Identity.Client/Extensibility/AcquireTokenForClientBuilderExtensions.cs

```diff
@@ -40,5 +42,35 @@ public static AcquireTokenForClientParameterBuilder WithProofOfPosessionKeyId(
 
             return builder;
         }
+
+        /// <summary>
+        /// Add extra body parameters to the token request. These parameters are added to the cache key to associate these parameters with the acquired token.
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="extrabodyparams">List of additional body parameters</param>
+        /// <returns></returns>
+        public static AcquireTokenForClientParameterBuilder WithExtraBodyParameters(
```

Comment:
Thx, can you also pls log a feature request to the same for WithExtraQueryParameters? i.e. associate tokens with those params?